### PR TITLE
DB-7870 allow user mapping for Kerberos and Native authentication

### PR DESCRIPTION
**Pull request by yxia92**
Please see comment from [DB-7870](
https://splicemachine.atlassian.net/browse/DB-7870?focusedCommentId=59600&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-59600) for details.

Please also review [PR](https://github.com/splicemachine/spliceengine-ee/pull/343) in the EE branch.

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
 	 <LI> CURRENT_USER
 	 <LI> CURRENT_ROLE
 	 <LI> SESSION_USER
+     <LI> GROUP_USER
```
Comment (MurrayBrown) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: nit: whitespace

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java [hunk 2]
```diff
@@ -170,7 +178,7 @@ public ValueNode bindExpression(FromList fromList,
 			dtd = DataTypeDescriptor.getBuiltInDataTypeDescriptor(
 				// size: 2+(2*128) start and end text quote plus max # of
```
Comment (MurrayBrown) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: this size is much larger, but the comment is unchanged?

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java [hunk 3]
```diff
@@ -254,6 +262,7 @@ public void generateExpression(ExpressionClassBuilder acb,
 		int argCount = 0;
 
 		if (methodName.equals("getCurrentRoleIdDelimited") ||
+				methodName.equals("getCurrentGroupUserDelimited") ||
```
Comment (MurrayBrown) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: nit: whitespace?

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java (hunk 2 above)
Comment (msirek) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: It's VARCHAR, so this is a max length, maybe to accommodate really long names.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java (hunk 1 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: fixed

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java (hunk 2 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: current_role could return names of multiple active roles, the original size is too small. So I just increased it to max.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java (hunk 3 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SpecialFunctionNode.java]: fixed

**Pull request merged**